### PR TITLE
docs: add ginger upgrade to docs, update links

### PR DESCRIPTION
**Pull request by jcstein**
## Summary by CodeRabbit

- **Documentation**
	- Updated guides for the Arabica devnet, Mainnet Beta, Mocha testnet, network upgrade process, and participation to reflect the transition from the "Lemongrass" to the "Ginger" network upgrade.
	- Enhanced clarity and detail in the network upgrade process documentation, including new coordination methods and a clearer upgrade sequence.
	- Expanded information on the Arabica devnet faucet and RPC endpoints, with warnings against relying on community endpoints for production.
	- Improved readability and structure in the Arabica devnet documentation, including detailed instructions for syncing blocks and accessing state through gRPC endpoints.

### CODE REVIEW

File: how-to-guides/network-upgrade-process.md [hunk 1]
```diff
@@ -4,55 +4,58 @@ description: Overview of the Celestia network upgrade process.
...
-  their binary to be network upgrade ready, the network upgrade does not happen right
-  away, but some short time in the future at a specific block number.
-- Testing of the features, which happens on testnets first prior to activating on
-  Mainnet Beta in order to ensure the network can upgrade securely.
+The upgrade process can be broken down into a few steps:
+1. [Celestia Improvement Proposals](https://cips.celestia.org) (CIPs) are created for consensus-breaking changes and features that impact user experience. These CIPs are included in a meta-CIP, which define the scope of the upgrade.
+1. Celestia core developer teams implement the features defined in the CIPs.
+1. A new binary is released with the new features to be tested on testnets.
+1. Validators upgrade their nodes to the new binary, on [Arabica devnet](./arabica-devnet.md), [Mocha testnet](./mocha-testnet.md), and finally on Celestia [Mainnet Beta](./mainnet.md).
+    - For upgrades using pre-programmed height (v2): A block number for when the breaking network upgrade occurs
```
Comment (rootulp) [inline how-to-guides/network-upgrade-process.md:25]:
```suggestion
    - Upgrades using pre-programmed height (v2) activate at a predetermined block number.
```

File: how-to-guides/network-upgrade-process.md [hunk 2]
```diff
@@ -4,55 +4,58 @@ description: Overview of the Celestia network upgrade process.
...
+    - For upgrades using in-protocol signaling (v3+): Reaching 5/6 validator consensus on the new version
```
Comment (rootulp) [inline how-to-guides/network-upgrade-process.md:26]:
```suggestion
    - Upgrades using in-protocol signaling (v3+) activate one week after 5/6 of the voting power has signaled for a particular version
```

File: how-to-guides/network-upgrade-process.md [hunk 3]
```diff
@@ -4,55 +4,58 @@ description: Overview of the Celestia network upgrade process.
...
+Key features include:
+- [CIP-21](https://cips.celestia.org/cip-21.html): Introduce blob type with verified signer
+- [CIP-24](https://cips.celestia.org/cip-24.html): Versioned Gas Scheduler Variables
+- [CIP-26](https://cips.celestia.org/cip-26.html): Versioned timeouts
+- [CIP-27](https://cips.celestia.org/cip-27.html): Block limits for number of PFBs and non-PFBs
+- [CIP-28](https://cips.celestia.org/cip-28.html): Transaction size limit
 
-- If you are a consensus node or validator operator: you will need to download and run a celestia-app binary >= v2.0.0 prior to the `--v2-upgrade-height` to remain on the canonical chain.
-- If you are a DA node operator, you will need to download and run a compatible celestia-node binary >= v0.16.0-rc0 prior to the upgrade height.
+Unlike the Lemongrass upgrade, there will not be a pre-programmed upgrade height. Instead, validators will signal their readiness for v3 through in-protocol signaling, and the upgrade will automatically activate once 5/6 of validators have signaled support.
```
Comment (rootulp) [inline how-to-guides/network-upgrade-process.md:41]:
```suggestion
Unlike the Lemongrass upgrade, there will not be a pre-programmed upgrade height. Instead, validators will signal their readiness for v3 through in-protocol signaling, and the upgrade will automatically activate one week after 5/6 of voting power have signaled for a particular version.
```

**Review by rootulp — APPROVED** [review]
LGTM with a few optional nits.

[optional] we could remove the **Network upgrades** section from arabica, mocha, mainnet beta pages b/c it is duplicative of the network upgrade process page.

**Comment by jcstein** [comment]
> [optional] we could remove the Network upgrades section from arabica, mocha, mainnet beta pages b/c it is duplicative of the network upgrade process page.

this is a good suggestion, I think I'll leave it in for discoverability, for now

**Pull request merged**